src/matching/MatchingReducerFactory.js

```diff
@@ -0,0 +1,100 @@
+import { matcher as defaultMatcher } from './matchers';
+import { warn } from '../utils/logger';
+
+const identity = value => value;
+
+/**
+ * @class MatchingReducerFactory
+ *
+ * A simple pattern matching abstraction
+ * for redux reducers.
+ */
+export default class MatchingReducerFactory {
+
+  /**
+   * @param {any} Initial app state
+   * @param {Matcher} Default Matcher implementation
+   */
+  constructor(initialAppState, defaultMatcherImpl = defaultMatcher) {
+    this.defaultMatcherImpl = defaultMatcherImpl;
+    this.initialAppState = initialAppState;
+    this.matchersWithHandlers = [];
+  }
+
+  /**
+   * Registers action handler
+   *
+   * @param {String} String pattenr
+   * @param {Function} Action Handler (reducer)
+   * @param {Matcher} [Optional] Specific Matcher implementation
+   */
+  case(pattern, actionHandler, matcherImpl) {
+    const matcher = matcherImpl ?
+      matcherImpl(pattern) :
+      this.defaultMatcherImpl(pattern);
+
+    this.matchersWithHandlers.push({
+      matcher,
+      actionHandler
+    });
+
+    return this;
+  }
+
+  /**
+   * Creates the Reducer
+   *
+   * @return {Function} plain old redux Reducer
+   */
+  toReducer() {
+    return (appState = this.initialAppState, action) => {
+      if (action) {
+        if (!action.type) {
+          // Logging is technically impurity yet this serves
+          // just for developer to eliminate any potential issues
+          // in development phase.
```

Comment:
I don't think it's desired here. Because this potentially leads to runtime error, so I don't think that "catching" this error and logging to console even in production would be anyhow dangerous.